src/containers/Tenant/Diagnostics/Overview/TableInfo/TableInfo.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useMemo} from 'react';
+import cn from 'bem-cn-lite';
+
+import type {TEvDescribeSchemeResult} from '../../../../../types/api/schema';
+
+import {InfoViewer, InfoViewerItem} from '../../../../../components/InfoViewer';
+
+import {getEntityName} from '../../../utils';
+
+import {prepareTableInfo} from './prepareTableInfo';
+
+import i18n from './i18n';
+
+import './TableInfo.scss';
+
+const b = cn('ydb-diagnostics-table-info');
+
+const checkIsEmptyTableInfo = (
+    generalTableInfo: InfoViewerItem[],
+    tabletMetricsInfo: InfoViewerItem[],
+    partitionConfigInfo: InfoViewerItem[],
+    tableStatsInfo: InfoViewerItem[][],
+) => {
+    return (
+        [generalTableInfo, tabletMetricsInfo, partitionConfigInfo, tableStatsInfo.flat()].flat()
+            .length === 0
+    );
+};
+
+interface TableInfoProps {
+    data?: TEvDescribeSchemeResult;
+}
+
+export const TableInfo = ({data}: TableInfoProps) => {
+    const entityName = getEntityName(data?.PathDescription);
+
+    const {
+        generalTableInfo = [],
+        tableStatsInfo = [],
+        tabletMetricsInfo = [],
+        partitionConfigInfo = [],
+    } = useMemo(() => prepareTableInfo(data), [data]);
+
+    if (
+        checkIsEmptyTableInfo(
+            generalTableInfo,
+            tabletMetricsInfo,
+            partitionConfigInfo,
+            tableStatsInfo,
+        )
```

Comment:
don't have any particular solution in mind, just notice that this still looks heavy
also — what if the set of fields determining the emptiness changes? This will entail changes here as well as in the function itself
i'd like to write something like `if (isEmpty(data)) {}` here and never touch it again
wouldn't have the fields calculated twice though, of course

Reply:
I discovered, that we don’t need such a check here. We can just add renderEmptyState on `generalTableInfo` block. In case if every field is empty, it will render a title, as every other `renderEmptyState` return null. Also it tackles the case, when `generalTableInfo` block is empty, while others aren’t (it will be displayed “No Table data” while there will be some data in other blocks)